libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -999,6 +999,13 @@ void AP_GPS::update_instance(uint8_t instance)
             tnow = state[instance].last_corrected_gps_time_us/1000U;
             state[instance].corrected_timestamp_updated = false;
         }
+
+        // we set delta_time_ms to the timeout value when clearing
+        // state; use it being zero to mark first message
+        if (timing[instance].delta_time_ms == GPS_TIMEOUT_MS) {
```

Comment:
this seems odd - what stops delta_time_ms from being GPS_TIMEOUT_MS later in the flight?
maybe a "bool announced_gps" variable would be better?

Reply:
Added a bool.